colmeiad/src/main.rs

```diff
@@ -0,0 +1,90 @@
+use async_std::{sync::RwLock, task};
+use colmeia_hypercore::*;
+use std::sync::Arc;
+use tide::{Request, StatusCode};
+
+fn name() -> String {
+    let args: Vec<String> = std::env::args().skip(1).collect();
+    args.first()
+        .expect("must have hash name as argument")
+        .into()
+}
+
+#[derive(serde::Serialize, Debug)]
+struct FeedInfo {
+    len: u64,
+    blocks: u64,
+}
+
+#[derive(serde::Serialize, Debug)]
+struct Info {
+    metadata: FeedInfo,
+    content: Option<FeedInfo>,
+}
+
+type State<Storage> = Arc<RwLock<Hyperdrive<Storage>>>;
+
+async fn get_info<Storage>(req: Request<State<Storage>>) -> tide::Result<tide::Response>
+where
+    Storage: random_access_storage::RandomAccess<Error = Box<dyn std::error::Error + Send + Sync>>
```

Comment:
TODO: Investigate https://github.com/datrs/hypercore/issues/109 impact here - likely would be simpler to use a `dyn Storage`